osu.Game/Screens/OnlinePlay/Multiplayer/Match/BeatmapSelectionControl.cs

```diff
@@ -47,7 +47,11 @@ private void load()
                         RelativeSizeAxes = Axes.X,
                         Height = 40,
                         Text = "Select beatmap",
-                        Action = () => matchSubScreen.Push(new MultiplayerMatchSongSelect()),
+                        Action = () =>
+                        {
+                            if (matchSubScreen.IsCurrentScreen())
+                                matchSubScreen.Push(new MultiplayerMatchSongSelect());
```

Comment:
This (and the similar addition below) are not required with the `IsLoading` check in place, but felt good for safety.